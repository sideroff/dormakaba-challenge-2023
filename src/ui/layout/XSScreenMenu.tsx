import { MouseEvent } from 'react';
import Box from '@mui/material/Box';
import IconButton from '@mui/material/IconButton';
import Menu from '@mui/material/Menu';
import MenuIcon from '@mui/icons-material/Menu';
import { NavBarMenuItem } from '@/ui/layout/NavBarMenuItem';

interface XSScreenMenuProps {
  navAnchorElement: null | HTMLElement;
  onNavMenuOpen: (event: MouseEvent<HTMLElement>) => void;
  onNavMenuClose: () => void;
}

export function XSScreenMenu({
  onNavMenuOpen,
  onNavMenuClose,
  navAnchorElement,
}: XSScreenMenuProps) {
  return (
    <Box sx={{ flexGrow: 1, display: { xs: 'flex', md: 'none' } }}>
      <IconButton
        size="large"
        aria-label="menu button"
        aria-controls="menu-appbar"
        aria-haspopup="true"
        onClick={onNavMenuOpen}
      >
        <MenuIcon />
      </IconButton>
      <Menu
        id="menu-appbar"
        anchorEl={navAnchorElement}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
        keepMounted
        transformOrigin={{
          vertical: 'top',
          horizontal: 'left',
        }}
        open={Boolean(navAnchorElement)}
        onClose={onNavMenuClose}
        sx={{
          display: { xs: 'block', md: 'none' },
        }}
      >
        <NavBarMenuItem onClick={onNavMenuClose} label="Doors" href="/doors" />
      </Menu>
    </Box>
  );
}
