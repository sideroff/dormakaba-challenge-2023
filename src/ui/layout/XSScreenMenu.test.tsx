import { render } from '@testing-library/react';
import { XSScreenMenu } from './XSScreenMenu';

jest.mock('next/router', () => require('next-router-mock'));

describe('XSScreenMenu', () => {
  it('should render correctly', () => {
    const handleNavMenuOpen = jest.fn();
    const handleNavMenuClose = jest.fn();

    const { container } = render(
      <XSScreenMenu
        navAnchorElement={null}
        onNavMenuOpen={handleNavMenuOpen}
        onNavMenuClose={handleNavMenuClose}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
