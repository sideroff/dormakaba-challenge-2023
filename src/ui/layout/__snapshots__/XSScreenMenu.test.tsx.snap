// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XSScreenMenu should render correctly 1`] = `
<div
  class="MuiBox-root css-1lvtzne"
>
  <button
    aria-controls="menu-appbar"
    aria-haspopup="true"
    aria-label="menu button"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
